azurerm/resource_arm_private_dns_mx_record_test.go

```diff
@@ -0,0 +1,397 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+)
+
+func TestAccAzureRMPrivateDnsMxRecord_basic(t *testing.T) {
+	resourceName := "azurerm_private_dns_mx_record.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMPrivateDnsMxRecord_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsMxRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsMxRecordExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsMxRecord_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_private_dns_mx_record.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateDnsMxRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateDnsMxRecord_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateDnsMxRecordExists(resourceName),
+				),
+			},
```

Comment:
We should do a sanity check that import works through each apply by adding the import test step between every test step.

Reply:
Well, any idea on how to implement this?

Reply:
Ah, my fault. I commented on the wrong line. We want
```
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```
on lines:

https://github.com/terraform-providers/terraform-provider-azurerm/pull/4915/files#diff-e8c9d50e92f6adf13be4cb73347f85e9R117

https://github.com/terraform-providers/terraform-provider-azurerm/pull/4915/files#diff-e8c9d50e92f6adf13be4cb73347f85e9R94

and

https://github.com/terraform-providers/terraform-provider-azurerm/pull/4915/files#diff-e8c9d50e92f6adf13be4cb73347f85e9R87